doc/example/gradient_check.ipynb

```diff
@@ -0,0 +1,903 @@
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>sd_pSTAT5A_rel</th>\n",
+       "      <td>-4.106435e+12</td>\n",
+       "      <td>-4.106388e+12</td>\n",
+       "      <td>-4.106482e+12</td>\n",
+       "      <td>-4.106435e+12</td>\n",
+       "      <td>9.372550e+07</td>\n",
+       "      <td>6.663599e+02</td>\n",
+       "      <td>1.622721e-10</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>sd_pSTAT5B_rel</th>\n",
+       "      <td>-2.467665e+11</td>\n",
+       "      <td>-2.468085e+11</td>\n",
+       "      <td>-2.467245e+11</td>\n",
+       "      <td>-2.467665e+11</td>\n",
+       "      <td>8.401874e+07</td>\n",
+       "      <td>3.362799e+01</td>\n",
+       "      <td>1.362745e-10</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>sd_rSTAT5A_rel</th>\n",
+       "      <td>3.684015e+01</td>\n",
+       "      <td>-4.769135e+07</td>\n",
+       "      <td>4.769149e+07</td>\n",
+       "      <td>7.324219e+01</td>\n",
+       "      <td>9.538284e+07</td>\n",
+       "      <td>3.640203e+01</td>\n",
+       "      <td>4.970090e-01</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "                              grad          fd_f          fd_b          fd_c  \\\n",
+       "Epo_degradation_BaF3  2.899805e+10  2.898349e+10  2.899516e+10  2.898933e+10   \n",
+       "k_exp_hetero         -1.822477e+03  8.247990e+07 -1.185836e+08 -1.805185e+07   \n",
+       "k_exp_homo            1.940159e+06 -7.634094e+06  2.620560e+07  9.285754e+06   \n",
+       "k_imp_hetero          1.324222e+09  9.636109e+08  1.401833e+09  1.182722e+09   \n",
+       "k_imp_homo            2.759777e+09  2.689595e+09  2.810697e+09  2.750146e+09   \n",
+       "k_phos               -3.183894e+10 -3.189761e+10 -3.181251e+10 -3.185506e+10   \n",
+       "sd_pSTAT5A_rel       -4.106435e+12 -4.106388e+12 -4.106482e+12 -4.106435e+12   \n",
+       "sd_pSTAT5B_rel       -2.467665e+11 -2.468085e+11 -2.467245e+11 -2.467665e+11   \n",
+       "sd_rSTAT5A_rel        3.684015e+01 -4.769135e+07  4.769149e+07  7.324219e+01   \n",
+       "\n",
+       "                            fd_err       abs_err       rel_err  \n",
+       "Epo_degradation_BaF3  1.166055e+07  8.729236e+06  3.011190e-04  \n",
+       "k_exp_hetero          2.010635e+08  1.805003e+07  9.998990e-01  \n",
+       "k_exp_homo            3.383970e+07  7.345596e+06  7.910607e-01  \n",
+       "k_imp_hetero          4.382216e+08  1.415004e+08  1.196396e-01  \n",
+       "k_imp_homo            1.211021e+08  9.630702e+06  3.501887e-03  \n",
+       "k_phos                8.509938e+07  1.611509e+07  5.058878e-04  \n",
+       "sd_pSTAT5A_rel        9.372550e+07  6.663599e+02  1.622721e-10  \n",
+       "sd_pSTAT5B_rel        8.401874e+07  3.362799e+01  1.362745e-10  \n",
+       "sd_rSTAT5A_rel        9.538284e+07  3.640203e+01  4.970090e-01  "
+      ]
+     },
```

Comment:
Do you think we should add some conclusion on this or one of the gradient checks? In the sense what would we think if we saw this gradient check: the absolute errors are rather high but can be expected with finite differences on random points. We see this also with fd_err, so changing the eps might make sense.
But what's reassuring is that the relative error is not too high in most cases.

Would redoing the gradient check with another eps make sense to show how it can affect the check?